hack/tests/e2e-ansible.sh

```diff
@@ -2,98 +2,136 @@
 
-DIR2="$(pwd)"
-# deploy the operator
-kubectl create -f deploy/service_account.yaml
-trap_add 'kubectl delete -f ${DIR2}/deploy/service_account.yaml' EXIT
-kubectl create -f deploy/role.yaml
-trap_add 'kubectl delete -f ${DIR2}/deploy/role.yaml' EXIT
-kubectl create -f deploy/role_binding.yaml
-trap_add 'kubectl delete -f ${DIR2}/deploy/role_binding.yaml' EXIT
-kubectl create -f deploy/crds/ansible_v1alpha1_memcached_crd.yaml
-trap_add 'kubectl delete -f ${DIR2}/deploy/crds/ansible_v1alpha1_memcached_crd.yaml' EXIT
-kubectl create -f deploy/operator.yaml
-trap_add 'kubectl delete -f ${DIR2}/deploy/operator.yaml' EXIT
-
-# wait for operator pod to run
-if ! timeout 1m kubectl rollout status deployment/memcached-operator;
-then
-    kubectl logs deployment/memcached-operator
-    exit 1
-fi
+OPERATORDIR="$(pwd)"
 
-# create CR
-kubectl create -f deploy/crds/ansible_v1alpha1_memcached_cr.yaml
-if ! timeout 20s bash -c -- 'until kubectl get deployment -l app=memcached | grep memcached; do sleep 1; done';
-then
-    kubectl logs deployment/memcached-operator
-    exit 1
-fi
-memcached_deployment=$(kubectl get deployment -l app=memcached -o jsonpath="{..metadata.name}")
-if ! timeout 1m kubectl rollout status deployment/${memcached_deployment};
-then
-    kubectl logs deployment/${memcached_deployment}
-    exit 1
-fi
+deploy_operator
+trap_add 'remove_operator' EXIT
+test_operator
+remove_operator
 
-# make a configmap that the finalizer should remove
-kubectl create configmap deleteme
-trap_add 'kubectl delete --ignore-not-found configmap deleteme' EXIT
+echo "###"
+echo "### Base image testing passed"
+echo "### Now testing migrate to hybrid operator"
+echo "###"
 
-kubectl delete -f ${DIR2}/deploy/crds/ansible_v1alpha1_memcached_cr.yaml --wait=true
-# if the finalizer did not delete the configmap...
-if kubectl get configmap deleteme;
-then
-    echo FAIL: the finalizer did not delete the configmap
-    kubectl logs deployment/memcached-operator
-    exit 1
-fi
+operator-sdk migrate
 
-# The deployment should get garbage collected, so we expect to fail getting the deployment.
-if ! timeout 20s bash -c -- "while kubectl get deployment ${memcached_deployment}; do sleep 1; done";
+if [[ ! -e build/Dockerfile.sdkold ]];
 then
-    kubectl logs deployment/memcached-operator
+    echo FAIL the old Dockerfile should have been renamed to Dockerfile.sdkold
     exit 1
 fi
 
+# We can't reliably run `dep ensure` because when there are changes to
+# operator-sdk itself, and those changes are not merged upstream, we hit this
+# bug: https://github.com/golang/dep/issues/1747
+# Instead, this re-uses operator-sdk's own vendor directory.
+cp -a $ROOTDIR/vendor ./
+mkdir -p vendor/github.com/operator-framework/operator-sdk/
+# We cannot just use operator-sdk from $GOPATH because compilation tries to use
+# its vendor directory, which can conflict with the local one.
+cp -a "$ROOTDIR"/{pkg,version,LICENSE} vendor/github.com/operator-framework/operator-sdk/
```

Comment:
`internal` will need copying as well. Is `LICENSE` necessary?

Reply:
Good call on `internal`. As for LICENSE I was just copying everything that `dep` retrieves. It probably doesn't matter, but I figured it wouldn't hurt to keep the vendor dir as close to what `dep` would normally create as possible.